cmd/podman/images/scp.go

```diff
@@ -0,0 +1,330 @@
+	podman := os.Args[0]
+	if v, found := os.LookupEnv("PODMAN_BINARY"); found {
+		podman = v // get path to podman bin
+	}
+	run := ""
+	if tag != "" {
+		return "", errors.Wrapf(define.ErrInvalidArg, "Renaming of an image is currently not supported")
+	}
+	run = podman + " image save " + imageFileOut + " --output=" + imageFileIn // run ssh image load of the file copied via scp. Files are reverse in thie case...
+	var buffer bytes.Buffer
+	session.Stdout = &buffer
+
+	if err := session.Run(run); err != nil { // run the command declared above
+		return "", err
+	}
+	n, err := scpD.CopyFrom(dial, imageFileIn, imageFileIn)
+	if err != nil {
+		fmt.Println(n, " Bytes copied before error")
+		return " ", err
+	}
+	return "", nil
+}
+
+// parseLocation returns the valid connection data based off of the information provided by the user
+// iden is a string array of ssh identity keys, url is a string array of paths to ssh clients. The first entry
+// in each array is the save location and the second entry is the load location if applicable. An empty string will be
+// inserted if identity is blank (not required in some cases).
+// serv is an array of service destinations
+func parseLocation(args []string, cfg *config.Config) ([]string, []string, map[string]config.Destination, error) {
+	iden := []string{}
+	url := []string{}
+	serv := map[string]config.Destination{}
+	switch len(args) {
+	case 2:
+		if strings.Contains(args[0], "::") {
+			if !(strings.Contains(args[1], "::")) { // if an image is specified, this mean we are loading from our client
+				scpOpts.Connections = append(scpOpts.Connections, args[0])
+				scpOpts.ToRemote = true
+			} else if strings.Contains(args[1], "::") { // both remote clients
+				scpOpts.FromRemote = true
+				scpOpts.ToRemote = true
+				if len(strings.Split(args[0], "::")) == 1 || (len(strings.Split(args[0], "::")) == 2 && len(strings.Split(args[1], "::")) == 2) { // is save->load w/ one image name or save-load w/ two image names
+					scpOpts.Connections = append(scpOpts.Connections, args[0])
+					scpOpts.Connections = append(scpOpts.Connections, args[1])
+				} else { // else its a load save (order of args)
+					scpOpts.Connections = append(scpOpts.Connections, args[1])
+					scpOpts.Connections = append(scpOpts.Connections, args[0])
+				}
+			}
+		} else if strings.Contains(args[1], "::") && len(scpOpts.Connections) < 2 { // if we are given image host::
+			scpOpts.ToRemote = true
+			scpOpts.Connections = append(scpOpts.Connections, args[1])
+		}
+	case 1:
+		if strings.Contains(args[0], "::") {
+			scpOpts.FromRemote = true
+			scpOpts.Connections = append(scpOpts.Connections, args[0])
+		}
+	}
+
+	for i, val := range scpOpts.Connections {
+		splitEnv := strings.SplitN(val, "::", 2)
+		switch len(splitEnv) {
+		case 1:
+			scpOpts.Connections[i] = splitEnv[0]
+		case 2:
+			scpOpts.ImageName = splitEnv[1]
+			//TODO: actually use the new name given by the user
+			scpOpts.Connections[i] = splitEnv[0]
+		}
```

Comment:
the TODO applies to case two which I have now changed to make more sense. Instead of len of the array (which is always 2) I am doing the length of the 2nd entry which will be 0 (no image) or more than 0, which means there is an image. The more than 0 option will be used for tag in the future.

Reply:
I can save the CLI input in the local array and then put it in connections